test/t8_IO/t8_gtest_vtk_reader.cxx

```diff
@@ -26,38 +26,78 @@ along with t8code; if not, write to the Free Software Foundation, Inc.,
 /**
  * This is currently a place-holder for a propper cmesh_vtk_reader-test.
  * The function is not implemented yet and therefore we do not provide a proper
- * test yet. 
+ * test yet. A proper test would compare the read files with a reference-cmesh. 
  * 
  */
-TEST (t8_cmesh_vtk_reader, dummy_test)
+
+
+/* *INDENT-OFF* */
+class vtk_reader : public testing::TestWithParam<vtk_file_type_t>{
+  protected:
+    void SetUp() override{
+      file_type = GetParam();
+      file = (int)file_type + 1;
+    }
+    int file;
+    vtk_file_type_t file_type;
+    const char* failing_files[3] = {
+      "no_file",
+      "non-existing-file.vtu",
+      "non-existing-file.vtp"
+    };
+    const char* test_files[3] = {
+      "no_file",
+      "test/testfiles/test_vtk_tri.vtu",
+      "test/testfiles/test_vtk_cube.vtp"
+    };
+    const int num_points[3] = {0, 121, 24};
+};
+/* *INDENT-ON* */
+
+/* All readers should fail properly with a non-existing file. */
+TEST_P (vtk_reader, vtk_to_cmesh_fail)
 {
 #if T8_WITH_VTK
   t8_cmesh_t          cmesh =
-    t8_cmesh_vtk_reader ("non-existing-file.vtu", 0, 0, sc_MPI_COMM_WORLD,
-                         VTK_FILE_ERROR);
-  EXPECT_TRUE (cmesh == NULL);
-
-  cmesh =
-    t8_cmesh_vtk_reader ("non-existing-file.vtu", 0, 0, sc_MPI_COMM_WORLD,
-                         VTK_UNSTRUCTURED_FILE);
-  EXPECT_TRUE (cmesh == NULL);
-
-  cmesh =
-    t8_cmesh_vtk_reader ("non-existing-file.vtp", 0, 0, sc_MPI_COMM_WORLD,
-                         VTK_POLYDATA_FILE);
+    t8_cmesh_vtk_reader (failing_files[file], 0, 0, sc_MPI_COMM_WORLD,
+                         file_type);
   EXPECT_TRUE (cmesh == NULL);
+#else
+#endif
```

Comment:
Is there a reason for this?

Reply:
Yes. We can call the vtk_reader functions even if we don't link with vtk. We should also check the return values in that case.